script.js

```diff
@@ -1,4 +1,127 @@
+// Equipment:
+// 2 dice (up to 7 dice for older players)
+// Score Chart for win loss
+
+// How to play
+
+// Roll the dice and put them in order to make the highest number possible. If you roll a 4 and an 6, for example, your best answer would be 64. Using 3 dice, a roll of 3, 5 and 2 should give you 532, and so on. Write down your answer, pass the dice, and challenge the next player to Beat That!
+
+// Play in rounds and assign a winner to each round.
+
+// Base
+
+// Requirements
+
+// There are 2 players and players take turns.
+
+// When a player clicks Submit, the game rolls 2 dice and shows the dice rolls, for example 3 and 6.
+
+// The player picks the order of the dice they want. For example, if they wanted the number 63, they would specify that the 2nd dice goes first. You can choose how the player specifies dice order.
+
+// After both players have rolled and chosen dice order, the player with the higher combined number wins.
+
+// 1st mode => 1st User runs game clicks submit => diceRoll twice => display results.
+// 2nd mode => chooses order of dice integer as strings? ex 6 and 3 would be 63. how? setting array numbers maybe.
+// 3rd mode => 2nd player runs game clicks submit => diceRoll twice => display results.
+// 4th mode => chooses order of dice integer as strings? ex 6 and 3 would be 63. how? setting array numbers maybe.
+// 5th step => compare results, if player 1 > player 2 or vice versa, the player with the highest number wins.
+
+// what do i need?
+
+// Create 1st game mode
+var GAME_STATE_DICE_ROLL = "GAME_STATE_DICE_ROLL";
+var GAME_STATE_CHOOSE_DICE_ORDER = "GAME_STATE_CHOOSE_DICE_ORDER";
+var GAME_STATE_COMPARE_SCORES = "GAME_STATE_COMPARE_SCORES";
+var gameState = GAME_STATE_DICE_ROLL;
+
+var currentPlayerRolls = [];
+
+var currentPlayer = 1;
+var allPlayerScore = [];
+
+// dice roll function
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  console.log(rollDice, "dice roll");
+  return diceNumber;
+};
```

Comment:
i know this is the template code we gave for rolling dice, and its fine to reuse it as long as you understand, but think of ways that you can optimise it in terms of length. this will come in handy for blackjack too to simplify the deck generation + shuffling function. 

one way here is combine everything into a single return statement as its still fairly simple, i.e. `return Math.floor(Math.random() * 6) + 1;`